level2/restaurants/kubo_restaurants.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+# Restaurant class
+class Restaurant
+  attr_reader :name, :items
+
+  def initialize(name)
+    @name = name
+    @items = []
+  end
+
+  def add_items(*items)
+    items.each { |item| @items << item }
+  end
+
+  def remove_item(item_name)
+    @items.reject! { |item| item.name == item_name }
+  end
+
+  def place_order(order_items, order_discount = 0)
+    order_discount = order_discount[:discount] if order_discount != 0
+    order_price = 0
+    order_items.each do |order_item|
+      checked_item = @items.select { |i| i.name == order_item[:name] }[0]
+      return DOUBLE_DISCOUNT_ERROR if order_item[:discount] && !order_discount.zero?
+      return UNAVAILABLE_ITEM_ERROR unless checked_item
```

Comment:
when you have early returns from a method, you want to do them in such a way that no unnecessary code is executed before the return:

```
# bad
def my_method(should_return)
  x = [1,2,3] # this variable is created but not used if should_return is true
  return if should_return
  x
end

# good
def my_method(should_return)
  return if should_return
  ...
end
```